pennylane/operation.py

```diff
@@ -323,6 +323,16 @@ def classproperty(func):
     return ClassPropertyDescriptor(func)
 
 
+# =============================================================================
+# Error classes
+# =============================================================================
+
+
+class GeneratorUndefinedError(Exception):
```

Comment:
I like this - we can use this pattern for all errors...

`DecompositionUndefinedError`
`MatrixUndefinedError`
`SparseMatrixUndefinedError`
`EigvalsUndefinedError`
...